src/thunderbots/software/ai/intent/dribble_intent.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "ai/intent/intent.h"
+#include "ai/primitive/dribble_primitive.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+
+class DribbleIntent : public Intent, public DribblePrimitive
+{
+   public:
+    static const std::string INTENT_NAME;
+    /**
+     * Creates a new Dribble Intent
+     * Moves the robot in a straight line between its current position and the given
+     * destination with the dribbler on
+     *
+     * @param robot_id The id of the Robot to run this Intent
+     * @param dest The final destination of the movement
+     * @param final_angle The final orientation the robot should have at the end
+     * of the movement
+     * @param final_speed The final speed the Robot should have when it reaches
+     * its destination at the end of the movement
+     * @param rpm The rotation speed of the dribbler in RPMs
+     * @param small_kick_allowed Boolean of whether the robot is allowed o do a small
+     * kick while dribbling in order to release the ball. This is due to the rule that
+     * a robot may not dribble more than 1 meter without releasing the ball. This is not
+     * obeyed in firmware
```

Comment:
How would this work? Would a `DribbleIntent` with this flag set cause the robot to immediately kick? Or would robots need to automatically figure out when to kick?

Reply:
Likely it's up to the robot. This flag is part of the current protocol but I'm not sure exactly how it's handled on the robot. For legacy purposes I'd like to leave it for now but define this better in the future.